programs/castle-vault/src/adapters/jet.rs

```diff
@@ -125,7 +136,15 @@ impl<'info> LendingMarket for JetAccounts<'info> {
                 context.with_signer(&[&self.vault.authority_seeds()]),
                 Amount::from_deposit_notes(amount),
             ),
-        }
+        }?;
+
+        let vault_reserve_value_delta = self.convert_amount_lp_to_reserve(amount)?;
+        let jet_value = self.vault.actual_allocations[Provider::Jet]
+            .value
+            .checked_sub(vault_reserve_value_delta)
+            .ok_or(ErrorCode::MathError)?;
+        self.vault.actual_allocations[Provider::Jet].update(jet_value, self.clock.slot);
```

Comment:
I also tried retrieving the value from `vault_jet_lp_token.amount`, but it contains the old value before running `jet::cpi::withdraw_tokens`. Same for solend and port.

I guess this is probably how SPL token accounts work? If you CPI to transfer SPL tokens and then check balance in the same instruction, you get the old value before the transfer because apparently it's not updated yet.